### PR TITLE
CB-3372 feat: codemirror6 autocomplete

### PR DESCRIPTION
**Pull request by Wroud**

### CODE REVIEW

File: webapp/packages/plugin-codemirror6/src/useEditorAutocompletion.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+import type { IEditorRef } from './IEditorRef';
+
+const styles = css`
+  :global(.cm-fix-position) {
+    margin-left: var(--cm-fix-position-left);
+    margin-top: var(--cm-fix-position-top);
+    z-index: 100;
+  }
+  :global(.cm-fixed-hack) {
```
Comment (devnaumov) [inline webapp/packages/plugin-codemirror6/src/useEditorAutocompletion.ts]: Please avoid using the word 'hack' in class (or variable) names. It is preferable to come up with a meaningful name and add a comment to provide clarity.

**Review by devnaumov — APPROVED** [review]

File: webapp/packages/plugin-sql-editor-new/src/SQLEditor/SQLCodeEditorPanel/SQLCodeEditorPanel.tsx [hunk 2]
```diff
@@ -21,6 +22,8 @@ import { ACTIVE_QUERY_EXTENSION } from '../ACTIVE_QUERY_EXTENSION';
 import { QUERY_STATUS_GUTTER_EXTENSION } from '../QUERY_STATUS_GUTTER_EXTENSION';
 import { SQLCodeEditorLoader } from '../SQLCodeEditor/SQLCodeEditorLoader';
 import { useSQLCodeEditor } from '../SQLCodeEditor/useSQLCodeEditor';
+import { useSqlDIalectAutocompletion } from '../useSqlDIalectAutocompletion';
```
Comment (tfokina) [inline webapp/packages/plugin-sql-editor-new/src/SQLEditor/SQLCodeEditorPanel/SQLCodeEditorPanel.tsx]: **DI**alect typo

File: webapp/packages/plugin-sql-editor-new/src/SQLEditor/useSqlDIalectAutocompletion.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    const word = context.matchBefore(/[\w*]*/);
+
+    if (word === null) {
+      return null;
+    }
+
+    try {
+      const proposals = await data.getHintProposals(context.pos, word.text, !context.explicit);
+
+      const workLowerCase = word.text.toLocaleLowerCase();
```
Comment (tfokina) [inline webapp/packages/plugin-sql-editor-new/src/SQLEditor/useSqlDIalectAutocompletion.ts]: wordLowerCase?

**Review by tfokina — APPROVED** [review]

File: webapp/packages/plugin-sql-editor-new/src/SQLEditor/SQLCodeEditorPanel/SQLCodeEditorPanel.tsx (hunk 2 above)
Comment (Wroud) [reply webapp/packages/plugin-sql-editor-new/src/SQLEditor/SQLCodeEditorPanel/SQLCodeEditorPanel.tsx]: fixed

File: webapp/packages/plugin-sql-editor-new/src/SQLEditor/useSqlDIalectAutocompletion.ts (hunk 3 above)
Comment (Wroud) [reply webapp/packages/plugin-sql-editor-new/src/SQLEditor/useSqlDIalectAutocompletion.ts]: fixed

File: webapp/packages/plugin-codemirror6/src/useEditorAutocompletion.ts (hunk 1 above)
Comment (Wroud) [reply webapp/packages/plugin-codemirror6/src/useEditorAutocompletion.ts]: fixed

File: webapp/packages/plugin-codemirror6/src/theme/_base-code-editor.scss [hunk 4]
```diff
@@ -38,66 +37,89 @@
...
   }
 
   .editor .cm-content ::selection {
     color: white;
   }
 
   .editor .cm-line::selection,
-  .editor .cm-line > :global(span)::selection,
-  .editor .cm-line > :global(span) > :global(span)::selection {
+  .editor .cm-line> :global(span)::selection,
```
Comment (devnaumov) [inline webapp/packages/plugin-codemirror6/src/theme/_base-code-editor.scss:69]: Seems like something wrong with format

**Review by devnaumov — APPROVED** [review]

**Pull request merged**